steps/generate_analytical_dataset.py

```diff
@@ -23,31 +26,45 @@ def main():
     collections = get_collections(secrets_response)
     published_database_name = get_published_db_name()
     database_name = get_staging_db_name()
-    spark = get_spark_session()
     tables = get_tables(database_name)
+
+
+    tag_values = []
+    collection_names = []
+    adg_hive_tables = []
+
```

Comment:
Could be more cleaner  if these properties are encapsulated in to a object which makes it more readable